drivers/docker/driver_test.go

```diff
@@ -2911,28 +2911,159 @@ func TestDockerDriver_memoryLimits(t *testing.T) {
+		{
+			name:            "not signal",
+			runtime:         runtime.GOOS,
+			specifiedSignal: "SIGDOESNOTEXIST",
+			expectedSignal:  "NA", // throws error
+		},
+	}
 
-	t.Run("default", func(t *testing.T) {
-		s, err := d.parseSignal(runtime.GOOS, "")
-		require.NoError(t, err)
-		require.Equal(t, syscall.SIGTERM, s)
-	})
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			s, err := parseSignal(tc.runtime, tc.specifiedSignal)
+			if tc.expectedSignal == "NA" {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				require.Equal(t, s.(syscall.Signal), s)
+			}
+		})
+	}
+}
 
-	t.Run("set", func(t *testing.T) {
-		s, err := d.parseSignal(runtime.GOOS, "SIGHUP")
-		require.NoError(t, err)
-		require.Equal(t, syscall.SIGHUP, s)
-	})
+// This test asserts that Nomad isn't overriding the STOPSIGNAL in a Dockerfile
+func TestDockerDriver_StopSignal(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip("Skipped on windows, we don't have image variants available")
+	}
+
+	testutil.DockerCompatible(t)
+	stopSigTaskCfg := newTaskConfig("stopsignal", []string{"sleep", "9001"})
+	defaultTaskCfg := newTaskConfig("", []string{"sleep", "9001"})
+	stopTask := &drivers.TaskConfig{
+		ID:        uuid.Generate(),
+		Name:      "stopsig-only",
+		AllocID:   uuid.Generate(),
+		Resources: basicResources,
+	}
+	require.NoError(t, stopTask.EncodeConcreteDriverConfig(&stopSigTaskCfg))
+
+	stopKillTask := &drivers.TaskConfig{
+		ID:        uuid.Generate(),
+		Name:      "stopsig-killsig",
+		AllocID:   uuid.Generate(),
+		Resources: basicResources,
+	}
+	require.NoError(t, stopKillTask.EncodeConcreteDriverConfig(&stopSigTaskCfg))
+
+	killTask := &drivers.TaskConfig{
+		ID:        uuid.Generate(),
+		Name:      "killsig-only",
+		AllocID:   uuid.Generate(),
+		Resources: basicResources,
+	}
+	require.NoError(t, killTask.EncodeConcreteDriverConfig(&defaultTaskCfg))
 
-	t.Run("windows conversion", func(t *testing.T) {
-		s, err := d.parseSignal("windows", "SIGINT")
+	defaultTask := &drivers.TaskConfig{
+		ID:        uuid.Generate(),
+		Name:      "nosigs-default",
+		AllocID:   uuid.Generate(),
+		Resources: basicResources,
+	}
+	require.NoError(t, defaultTask.EncodeConcreteDriverConfig(&defaultTaskCfg))
+
+	type taskInfo struct {
+		killSig      string
+		expectedSigs []string
+	}
+
+	taskList := map[*drivers.TaskConfig]taskInfo{
+		stopTask:     {killSig: "", expectedSigs: []string{"19", "9"}},
+		defaultTask:  {killSig: "", expectedSigs: []string{"15", "9"}},
+		stopKillTask: {killSig: "SIGTERM", expectedSigs: []string{"15", "19", "9"}},
+		killTask:     {killSig: "SIGTERM", expectedSigs: []string{"15", "15", "9"}},
+	}
+
+	d := dockerDriverHarness(t, nil)
+
+	for task := range taskList {
+		cleanup := d.MkAllocDir(task, true)
+		defer cleanup()
+
+		if task.Name == "stopsig-only" || task.Name == "stopsig-killsig" {
+			copyImage(t, task.TaskDir(), "busybox_stopsignal.tar") // Default busybox image with STOPSIGNAL 19 added
```

Comment:
I'd suggest adding a comment in `test-resources/docker` on how the image got created - or a bash script for regenerating it.